aws/amazon-freertos
# esp32: wifi-provisioning + support with NetworkManager

**Description**
-----------
Add wifi provisioning implementation for esp32:
- `iot_softap_wifi_provisioning(.h/.c)` implements a wifi-provisioning API that's easily compatible with existing `NetworkManager`
- `NetworkManager` can be wifi-provisioned by [Espressif's SoftAP Provisioning App](https://play.google.com/store/apps/details?id=com.espressif.provsoftap&hl=en_US&gl=US) by defining `IOT_WIFI_ENABLE_SOFTAP_PROVISIONING == 1`

**Testing:**
1. Define `IOT_WIFI_ENABLE_SOFTAP_PROVISIONING == 1` in your `aws_demo_config.h`
2. Configure, build, then run the demo on your ESP32
3. Download the [Espressif's SoftAP Provisioning App](https://play.google.com/store/apps/details?id=com.espressif.provsoftap&hl=en_US&gl=US)
4. Reset your board
5. Use and follow the instructions in the _Espressif SoftAP Provisioning App_. 
6. MCU attempts connection with provisioned SSID/pasword. If successful, the creds are stored for reuse in flash. Otherwise, the provision process fails.

Note, you can locally alter the `IotWifiSoftAPProv_Init(...)` to call `prvDeleteAllSavedNetworks()` at its beginning if you want to run the SoftAP provision flow again.


Checklist:
----------


- [X] My code is Linted.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
